### DISCUSSION TITLE
Secondary Account

### DISCUSSION THREAD

**Discussion by psybernoid** [ad-ha/mg-saic-ha#74] [General]
Does this integration require you use the primary account the car is paired to, or is setting up a secondary account, granting access to that via the app then using this integration with said secondary account sufficient?

**Comment by ad-ha** [comment]
Hi @psybernoid 

Once you get the second account configured, you should be able to access the integration with either account.

**Reply by psybernoid** [reply]
Yes. That worked a teat. Secondary account works fine with it.

**Reply by ad-ha** [reply]
Glad it is working.

Cheers

**Comment by KaherdinTristan** [comment]
**Hi**
First of all; this is the best MG SAIC integration I've tested for Home Assistant. GREAT work, please keep it up and update it as long as possible! 

**Summary**
I have shared two cars with one and the same Secondary Account.
The Integration can detect both cars by VIN numbers, but... 
...I seem to be unable to _add_ both (two) cars within the Integration and the Integration only allows me to show data from Car 2. 

**My guess**
I feel as if the MG SAIC servers will only allow the Secondary Account to access data from the most recent Car. I.e. I cant  access  data from two cars simultaneously using ONE Secondary Account.   

My "issue" regarding secondary account is as follows;
My family as two MG EV cars. 
1. MG4
2. MG ZS 

**Here we go...**
I just recently found out about the option to grant access from the iSmart app to a "secondary account". 
I created one (1) secondary account, lets call this the Secondary Account. 

**Car 1**
In the iSmart app, I granted the Secondary Account access to car 1.
I then ran the setup for this integration in Home Assistant, and got all the data from car 1 into Home Assistant. 
This created "Integration entity" 1

**Car 2**
In the iSmart app, I granted the same Secondary Account access to car 2.
I then ran the setup for this integration in Home Assistant. During the setup process I was prompted with a dialogue box asking me to pick from two VIN numbers (car 1 and car 2). I selected car 2.   
The setup process continued and I got all the data from car 2 into Home Assistant. 
This created "Integration entity" 2

The data for both Car 1 and Car 2 showed for three hours. 

**Error**
But three hours later the integration stopped getting data in Home Assistant (right side, "Logbook"). 
After trying to reload the integration, I removed both "Integration entity" 1 and 2. 

In the Integration I ran "add Entry" again. 
I was again prompted with the dialogue box to choose from two VIN numbers. 
Adding Car 2 works fine, and this created an "Integration entity" for Car 2,
But when trying to re-add Cars 1, I select the VIN number for Car 1, but the Integration still just adds a new instance of Car 2 (a new "Integration entity" is created, it holds the VIN number of the selected car, Car 1). But inside this entity the name is of the VIN number of Car 2. 

I will try to mitigate the issue by creating Unique Secondary Accounts for each of the cars (Car 1 and 2). 
But I hope others facing such issues can find this thread in the future and that they then might be helped out by me writing this down.    

_Best regards from Sweden_

**Comment by sanvalerius** [comment]
Hi @ad-ha ,
Thanks for this integration, it was exactly what I was looking for my MG Marvel R !

On the main readme you wrote:
`Using this integration causes the MG/SAIC mobile app to shut down, as per API requirements, since only one device can be connected at a time. Logging into the mobile app causes this integration to disconnect and fail to set up.`

I can suggest that you change to "This integration requires to setup a secondary account", has it seems to work like a charm.
Steps:
- Via the iSmart App, logout and create new secondary account(s) 
- On the iSmart application, Login back with primary account, then Go to Settings - Secondary Account
- Click "+" to create a secondary account, enable the "permanent" option, leave "Family" authorisation, put a name and a secondary email address.
- On the integration, select email and enter the secondary email defined on previous step.

I created two secondary accounts, and can use the mobile app + myhomeassistant + mybetahomeassistant toghether.

Best Regards

**Reply by ad-ha** [reply]
Hi @sanvalerius

Thanks for pointing that out. The secondary account just came out about a month ago and I forgot to update the Readme with that.

I will do so for the next update.

Glad that this is working fine for you. 

Cheers

**Comment by KaherdinTristan** [comment]
I just want to give a short update on my experience 

We have two MG EVs, and each car has a unique "secondary account" user. These "secondary account" users are then used by the Integration. 

This works like a charm.
Except for the minor issue that upon Integration start, one of two entities (cars) always fails to setup. And this issue is always mitigated my just reloading that entity.

**Reply by townsmcp** [reply]
If this is still a problem for you @KaherdinTristan, please check out and try the latest beta '1.1.0-beta1 - Multi car beta' - it should resolve the 2 car issue

**Comment by townsmcp** [comment]
Closing this discussion as the integration handles multiple cars,  either via 1 secondary account or multiple secondary accounts
